Frameworks/include/CollectionHelpers.h

```diff
@@ -75,10 +81,39 @@ HRESULT WRLToNSCollection<ABI::Windows::Foundation::Collections::IMap<HSTRING, H
     return S_OK;
 }
 
+// Specified for converting IPropertyValue of String to NSString for Toast Action conversion
+// TODO 8218419: Cannot currently consume values in Objective C projection without first converting to IPropertyValue
+template <>
+HRESULT WRLToNSCollection<ABI::Windows::Foundation::Collections::IPropertySet>(ABI::Windows::Foundation::Collections::IPropertySet* map,
+                                                                               NSDictionary* __autoreleasing* pDictionary) {
+    using NSType = Private::CollectionType<ABI::Windows::Foundation::Collections::IPropertySet>::NSMutableEquivalentType;
+    NSType* collection = [[NSType alloc] init];
+    HRESULT hr = WRLHelpers::ForEach(
+        map,
+        [&collection](const Microsoft::WRL::ComPtr<ABI::Windows::Foundation::Collections::IKeyValuePair<HSTRING, IInspectable*>>& pair,
+                      boolean* stop) {
+            Microsoft::WRL::Wrappers::HString key, value;
+            Microsoft::WRL::ComPtr<IInspectable> inspectableValue;
+            Microsoft::WRL::ComPtr<ABI::Windows::Foundation::IPropertyValue> propertyValue;
```

Comment:
put declarations as close as possible to their use. This isn't straight C where you need to put stack variables first or something weird. 

By placing the decl near its use, you are minimizing that amount of mental jumping a reader needs to do to remember what a variable's type is etc. 

Typically I also like to separate decl+use blocks by a newline to add to readability so it ends up as:

WRL::Foo foo;
RETURN_IF_FAILED(bacon->get_foo(&foo));

WRL::Wafffle waffle;
RETURN_IF_FAILED(foo->get_waffle(&waffle));